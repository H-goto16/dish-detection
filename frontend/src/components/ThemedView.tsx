import { View, type ViewProps } from 'react-native';

import useColorScheme from '@/hooks/useColorScheme';

export type ThemedViewProps = ViewProps & {
  lightColor?: string;
  darkColor?: string;
};

export const ThemedView = ({
  style,
  lightColor,
  darkColor,
  ...otherProps
}: ThemedViewProps) => {
  const colors = useColorScheme();

  return (
    <View
      style={[{ backgroundColor: colors.background }, style]}
      {...otherProps}
    />
  );
};
