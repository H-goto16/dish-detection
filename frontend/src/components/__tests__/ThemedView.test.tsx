import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import { ThemedView } from '../ThemedView';

// Mock useColorScheme hook
jest.mock('@/hooks/useColorScheme', () => ({
  __esModule: true,
  default: jest.fn(() => ({
    background: '#ffffff',
    text: '#000000',
    theme: 'light',
  })),
}));

describe('ThemedView', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ThemedView />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with children', () => {
    const tree = renderer
      .create(
        <ThemedView>
          <Text>Test content</Text>
        </ThemedView>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const component = renderer.create(<ThemedView />);
    expect(component).toBeTruthy();
  });

  it('renders with custom style', () => {
    const customStyle = {
      padding: 10,
      margin: 5,
    };

    const tree = renderer.create(<ThemedView style={customStyle} />).toJSON();

    expect(tree).toBeTruthy();
    if (tree && !Array.isArray(tree)) {
      expect(tree.props.style).toEqual([
        { backgroundColor: '#ffffff' },
        customStyle,
      ]);
    }
  });

  it('renders with testID prop', () => {
    const tree = renderer
      .create(<ThemedView testID='themed-view-test' />)
      .toJSON();

    expect(tree).toBeTruthy();
    if (tree && !Array.isArray(tree)) {
      expect(tree.props.testID).toBe('themed-view-test');
    }
  });

  it('passes through other props', () => {
    const tree = renderer
      .create(<ThemedView accessible={true} accessibilityLabel='Test view' />)
      .toJSON();

    expect(tree).toBeTruthy();
    if (tree && !Array.isArray(tree)) {
      expect(tree.props.accessible).toBe(true);
      expect(tree.props.accessibilityLabel).toBe('Test view');
    }
  });
});
