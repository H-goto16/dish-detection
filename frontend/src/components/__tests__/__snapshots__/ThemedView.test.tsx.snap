// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedView renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
      },
      undefined,
    ]
  }
/>
`;

exports[`ThemedView renders with children 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
      },
      undefined,
    ]
  }
>
  <Text>
    Test content
  </Text>
</View>
`;
